eva-remapping-ingest/src/test/java/uk/ac/ebi/eva/remapping/ingest/batch/io/RemappedSubmittedVariantsWriterTest.java

```diff
@@ -0,0 +1,194 @@
+package uk.ac.ebi.eva.remapping.ingest.batch.io;
+
+import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
+import com.lordofthejars.nosqlunit.mongodb.MongoDbConfigurationBuilder;
+import com.lordofthejars.nosqlunit.mongodb.MongoDbRule;
+import com.mongodb.MongoClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ApplicationContext;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.commons.accession.hashing.SHA1HashingFunction;
+
+import uk.ac.ebi.eva.accession.core.model.ISubmittedVariant;
+import uk.ac.ebi.eva.accession.core.model.SubmittedVariant;
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantEntity;
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantOperationEntity;
+import uk.ac.ebi.eva.accession.core.summary.SubmittedVariantSummaryFunction;
+import uk.ac.ebi.eva.remapping.ingest.batch.listeners.RemappingIngestCounts;
+import uk.ac.ebi.eva.remapping.ingest.test.configuration.BatchTestConfiguration;
+import uk.ac.ebi.eva.remapping.ingest.test.rule.FixSpringMongoDbRule;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+import static uk.ac.ebi.eva.remapping.ingest.configuration.BeanNames.REMAPPED_SUBMITTED_VARIANTS_WRITER;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {BatchTestConfiguration.class})
+@TestPropertySource("classpath:ingest-remapped-variants.properties")
+@UsingDataSet(locations = {"/test-data/submittedVariantEntity.json"})
+public class RemappedSubmittedVariantsWriterTest {
+
+    private static final String TEST_DB = "test-ingest-remapping";
+
+    @Autowired
+    private MongoClient mongoClient;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    @Qualifier(REMAPPED_SUBMITTED_VARIANTS_WRITER)
+    private RemappedSubmittedVariantsWriter writer;
+
+    private Function<ISubmittedVariant, String> hashingFunction;
+
+    //Required by nosql-unit
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Rule
+    public MongoDbRule mongoDbRule = new FixSpringMongoDbRule(
+            MongoDbConfigurationBuilder.mongoDb().databaseName(TEST_DB).build());
+
+    @Autowired
+    private RemappingIngestCounts remappingIngestCounts;
+
+    @Before
+    public void setUp() {
+        hashingFunction = new SubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
```

Comment:
**Optional:** have you tried initializing this outside of setup? not sure if this is necessarily a part of test setup...

Reply:
done